web/modules/weather_data/src/Service/WeatherAlertTrait.php

```diff
@@ -57,27 +57,65 @@ public static function tryParsingDescriptionText($str)
     /**
      * Get active alerts for a WFO grid cell.
      */
-    public function getAlertsForGrid($wfo, $x, $y)
+    public function getAlertsForGrid($wfo, $x, $y, $self = false)
     {
-        $geometry = $this->getGeometryFromGrid($wfo, $x, $y);
-        $point = $geometry[0];
+        if (!$self) {
+            $self = $this;
+        }
 
-        return $this->getAlertsForLatLon($point->lat, $point->lon);
-    }
+        $CACHE_KEY = "alerts $wfo/$x/$y";
+        $cache = $this->cache->get($CACHE_KEY);
+        if ($cache) {
+            return $cache->data;
+        }
 
-    /**
-     * Get active alerts for a latitude and longitude.
-     */
-    public function getAlertsForLatLon($lat, $lon)
-    {
-        $lat = round($lat, 4);
-        $lon = round($lon, 4);
+        $geometry = $self->getGeometryFromGrid($wfo, $x, $y);
+        $place = $self->getPlaceFromGrid($wfo, $x, $y);
+        $timezone = $place->timezone;
+
+        $alerts = $self->getFromWeatherAPI(
+            "/alerts/active?status=actual&area=$place->state",
+        )->features;
+
+        $geometry = array_map(function ($point) {
+            return $point->lon . " " . $point->lat;
+        }, $geometry);
+        $geometry = implode(",", $geometry);
```

Comment:
A future refactor would be to have `getGeometryFromGrid` just return the array of points instead of massaging them into a funky object. What's also kind of annoying is that in both places we get this geometry, we still have to transform it into a [WKT representation](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry) anyway.

Maybe it could be `getGeometryWKTFromGrid` instead... 🤔